### PR TITLE
Removing the colon block that tosses out paths that are not device path.

### PR DESCRIPTION
**Pull request by maryamariyan**
Fixes: dotnet/corefx#26359

Will create the corefx PR updating System.Runtime.Extensions shortly.
FYI, WIP: https://github.com/maryamariyan/corefx/pull/21/files

### CODE REVIEW

File: src/mscorlib/shared/System/IO/Path.Windows.cs [hunk 1]
```diff
@@ -51,38 +51,11 @@ public static string GetFullPath(string path)
...
-            // We don't want to check invalid characters for device format- see comments for extended above
-            string fullPath = PathHelper.Normalize(path, checkInvalidCharacters: !isDevice, expandShortPaths: true);
-
-            if (!isDevice)
-            {
-                // Emulate FileIOPermissions checks, retained for compatibility (normal invalid characters have already been checked)
-                if (PathInternal.HasWildCardCharacters(fullPath))
-                    throw new ArgumentException(SR.Argument_InvalidPathChars, nameof(path));
-            }
+            string fullPath = PathHelper.Normalize(path, checkInvalidCharacters: false, expandShortPaths: true);
```
Comment (jkotas) [inline src/mscorlib/shared/System/IO/Path.Windows.cs]: `PathHelper.Normalize` seems to be always called with the same 2nd and 3rd argument. Strip these arguments?

File: src/mscorlib/shared/System/IO/Path.Windows.cs [hunk 2]
```diff
@@ -51,38 +51,11 @@ public static string GetFullPath(string path)
...
 
             return fullPath;
```
Comment (jkotas) [inline src/mscorlib/shared/System/IO/Path.Windows.cs]: Nit: You can do direct `return PathHelper.Normalize(...`. No need to have local anymore.

File: src/mscorlib/shared/System/IO/Path.Windows.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/mscorlib/shared/System/IO/Path.Windows.cs]: I'm ok with stripping them. I hope to eventually provide an option on the public API around expanding short paths. canonicalization of paths is the dream that would take care of it, but it is incredibly hard to design/implement.

**Comment by maryamariyan** [comment]
I haven't finished updating all the tests that will fail by this change yet. 
So far System.Runtime.Extensions and System.IO.FileSystem are affected by this.
Will add a comment here once that is done.

**Comment by JeremyKuhne** [comment]
I've included @maryamariyan's changes in https://github.com/dotnet/coreclr/pull/16478. (Thanks @maryamariyan!) Closing this one.

**Pull request closed without merging**